### PR TITLE
[KOGITO-1031] Add flag to ignore self-signed SSL certificate

### PR DESCRIPTION
**Pull request by Kevin-Mok**
See: https://issues.redhat.com/browse/KOGITO-1031

This is an accompanying PR to [the one opened in kogito-pipelines](https://github.com/kiegroup/kogito-pipelines/pull/46). 

## Testing
I was able to [run the `update-maven-artifacts.py` script](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/kogito/job/kogito-images-deploy-1031/14/execution/node/66/log/) with the Maven test repository with the self-signed certificate, despite the given warnings (it would previously fail), using the code from this PR.

## Requirements
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by spolti — APPROVED** [review]

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -132,8 +135,8 @@ pipeline {
                 script {
                     // Update artifacts
                     updateArtifactCmd = 'python3 scripts/update-maven-artifacts.py'
-                    if (getMavenArtifactRepository() != '') {
-                        updateArtifactCmd += " --repo-url ${getMavenArtifactRepository()}"
+                    if(getMavenArtifactRepository() != ''){
```
Comment (Kaitou786) [inline Jenkinsfile.deploy]: Why did you remove spaces? I see that the space after `if` and before `{` is everywhere in this file, can you change back to follow the pattern?

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -101,6 +101,9 @@ pipeline {
             }
         }
         stage('Prepare for PR') {
+            when {
+                expression { return isRelease() }
+            }
```
Comment (radtriste) [inline Jenkinsfile.deploy]: leftover of rebasing. we now prepare for PR also in nightlies to update Maven artifacts
```suggestion
```

Comment (Kevin-Mok) [reply Jenkinsfile.deploy]: I have to add this check instead:
```groovy
when {
    expression { return params.GIT_AUTHOR == 'kiegroup' }
}
```

This is because if there is a matching branch name for the original PR, it should pass in that branch to build for images as well. But then, the BSIG bot will try to fork the user's repo when it has already forked the kiegroup repo and will throw [this error](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/custom/job/kmok/job/KOGITO-1031-full/job/kogito-images-deploy/6/execution/node/66/log/):
```
+ hub fork --remote-name=origin
Error creating fork: ****/kogito-images already exists on github.com
```

So, I'll add in this check to make sure the forking is valid.

Comment (radtriste) [reply Jenkinsfile.deploy]: Oh I see now ... then you should set `isRelease() || isCreateChangesPR()` instead, like for `Create PR` stage => https://github.com/kiegroup/kogito-images/blob/master/Jenkinsfile.deploy#L285

**Review by radtriste — APPROVED** [review]

**Comment by spolti** [comment]
@Kaitou786 are you ok with the latest changes?

**Comment by spolti** [comment]
@Kevin-Mok target version on jira is set to 1.1.0, can you please confirm it?
I not sure, but I don't think that new commits will be on 1.1 at this point.

**Comment by Kevin-Mok** [comment]
@spolti That's fine. This can be merged along with all the [other KOGITO-1031 PR's](https://github.com/kiegroup/kogito-pipelines/pull/46) which will likely be after 1.1 anyway.

**Review by Kaitou786 — APPROVED** [review]

**Comment by Kaitou786** [comment]
> @Kaitou786 are you ok with the latest changes?

Yup :)
